torch/onnx/_internal/fx/exporter.py

```diff
@@ -1058,92 +1059,64 @@ def save_model_with_external_data(
-    symbolic_fn: onnxscript.OnnxFunction,
+    symbolic_fn: Union[onnxscript.OnnxFunction, Callable],
     torch_args: tuple,
     torch_kwargs: dict,
 ):
     """Validate the op between ONNX Runtime and PyTorch."""
     # op-level validation
     # Symbolic_fn should have the same output as node.target (torch ops)
+    # TODO: torch.dtype to onnx.dtype conversion
+    # trace_only function is regular python function
+    function_name = (
+        symbolic_fn.name
+        if isinstance(symbolic_fn, onnxscript.OnnxFunction)
+        else symbolic_fn.__name__
+    )
     try:
         with evaluator.default_as(evaluator.ort_evaluator):
             expected_outputs = node.target(*torch_args, **torch_kwargs)  # type: ignore[operator]
             # TODO(titaiwang): Expose _convert_tensor_to_numpy and _convert_kwargs_for_onnx?
-            input_onnx = [_convert_tensor_to_numpy(x) for x in torch_args]
+            input_onnx = [
+                onnx_proto_utils._convert_tensor_to_numpy(x) for x in torch_args
+            ]
             # deal with dtype and device
-            kwargs_onnx = _convert_kwargs_for_onnx(torch_kwargs)
+            kwargs_onnx = _filter_incompatible_and_dtype_convert_kwargs(torch_kwargs)
             ort_outputs = symbolic_fn(*input_onnx, **kwargs_onnx)
 
-            for ort_output, expected_output in zip(ort_outputs, expected_outputs):
+            # TODO: add pytree structure comparison.
+            flattened_torch_outputs, _ = _pytree.tree_flatten(expected_outputs)
+            flattened_function_outputs, _ = _pytree.tree_flatten(ort_outputs)
+
+            assert flattened_torch_outputs
+            assert len(flattened_torch_outputs) == len(flattened_function_outputs)
+
+            for torch_output, function_output in zip(
+                flattened_torch_outputs, flattened_function_outputs
+            ):
                 try:
+                    if not isinstance(function_output, np.ndarray):
+                        # An onnxscript tensor
+                        function_output = function_output.value
+
+                    # Use torch.testing as opposed to np.testing to ensure dtypes and shapes match
                     torch.testing.assert_close(
-                        expected_output.numpy(),
-                        ort_output,
-                        check_device=False,
-                        atol=10e-4,
-                        rtol=10e-3,
+                        torch.tensor(function_output).cpu(),
+                        torch_output.cpu()
+                        if isinstance(torch_output, torch.Tensor)
+                        else torch.tensor(torch_output).cpu(),
+                        rtol=1e-4,
+                        atol=1e-3,
                     )
+
                 except AssertionError as e:
                     warnings.warn(
-                        f"Suppressed AssertionError:\n{e}.\n"
+                        f"\nSuppressed AssertionError:\n{e}.\n"
```

Comment:
nit: why start with a newline?

Reply:
I wanted to separate the warning from each op better. It was super crowded, and I couldn't tell the message was from which op. Ideally, there shouldn't be too many ops popping out warning messages.